backend/app/lm/handlers.py

```diff
@@ -102,33 +103,29 @@ async def process_request(self) -> Resp:
         config_event = log_lm_config(ses, usr, arena_request_event, config)
         # Build the request
         lm_request = await self.lm_request().evaluate(session=self.session)
+        print('messagges_prompt',lm_request.content.messages[0])
+        print('messagges_doc',lm_request.content.messages[1])
         lm_request_event = arena_request_event
+        mapping_list = []   
         # Do the masking
         if config.pii_removal:  # TODO an IF op could be added to build conditional delayed computations if needed
             if config.pii_removal == "masking":
                 async with create_task_group() as tg:
                     for message in lm_request.content.messages:
-
                         async def set_content(message=message):
-                            message.content = await masking(
-                                message.content
-                            ).evaluate(session=self.session)
-
+                            message.content = await masking(message.content).evaluate(session=self.session)
                         tg.start_soon(set_content)
             if config.pii_removal == "replace":
                 async with create_task_group() as tg:
                     for message in lm_request.content.messages:
-
                         async def set_content(message=message):
-                            message.content, _ = await replace_masking(
-                                message.content
-                            ).evaluate(session=self.session)
-
+                            message.content, _ = await replace_masking(message.content).evaluate(session=self.session)
+                            print("mapping", _)
+                            print("message.content", message.content)
+                            mapping_list.append(_)
```

Comment:
the convention we can adopt for now (and please add a comment to explicit it) is that the content that needs to be replaced back is only the one in messages whose `role` is `user`. For system messages, we want to anonymyze but we do not want to replace back as we may leak some info of the examples to the user.